nixos/modules/config/console.nix

```diff
@@ -8,14 +8,22 @@ let
 
   makeColor = i: concatMapStringsSep "," (x: "0x" + substring (2*i) 2 x);
 
-  isUnicode = hasSuffix "UTF-8" (toUpper config.i18n.defaultLocale);
+  isUnicode = '' \
+    LOCALE_ARCHIVE=${config.i18n.glibcLocales}/lib/locale/locale-archive \
+    LANG=${config.i18n.defaultLocale} \
+    ${pkgs.locale}/bin/locale -k identification-codeset | grep -i UTF-8 \
```

Comment:
add locale to nativeBuildInputs

Reply:
I think I'm not making a derivation here...

Reply:
This snippet is used in optimizedKeymap which uses runCommand which already has a nativeBuildInputs and preLVMCommands which has environment.systemPackages